### PR TITLE
Added the ability to handle parsing of multiline string values.

### PR DESCRIPTION
**Pull request by kawhites**
Rapidjson does not support multiline string values. This poses an issue for upgrade paths, where past Marketplace Content has such multiline strings AND where we switch loading code over to use rapidjson (e.g. switch from JsonSchema to Cereal does that switch to rapidjson). This PR adds support for parsing multiline string values.

### CODE REVIEW

File: include/rapidjson/reader.h
Comment (thomaslgdskyboxlabs) [inline include/rapidjson/reader.h:156]: weird indent

File: include/rapidjson/reader.h
Comment (thomaslgdskyboxlabs) [inline include/rapidjson/reader.h:975]: what about `if (parseFlags & kParseInsituFlag)` ?

File: include/rapidjson/reader.h
Comment (tylaing) [reply include/rapidjson/reader.h:156]:
```suggestion
	kParseMultiLineStringFlag = 512, //!< Allow parsing multi-line string values.
```
minor: better name for the intent of this flag?

also super minor: added a period at end of the code comment to match the other lines above it.

File: include/rapidjson/reader.h
Comment (tylaing) [inline include/rapidjson/reader.h:1044]: Not sure if we need to support other line endings like `\r` - how does the previous json loading code handle that?

File: include/rapidjson/reader.h
Comment (tylaing) [inline include/rapidjson/reader.h:1049]: What about other whitespace such as ' ' space character? Maybe Creator used spaces instead of tabs - again: how did previous json loading handle that?

**Review by tylaing — COMMENTED** [review]
A minor suggestion and a couple of questions to dig into.

File: include/rapidjson/reader.h
Comment (kawhites) [reply include/rapidjson/reader.h:156]: The weird indent isn't showing up locally for me

Comment (tylaing) [reply include/rapidjson/reader.h:156]: Indent looks fine to me too; I use a chrome plugin for GitHub that actually fixes the indent alignment to show accurately (whereas GitHub by default doesn't), give Refined GitHub a try if you'd like, I do recommend changing a couple of settings:

Disable the following in extension's Options:
- clean-conversation-headers (so that the target branch of a PR can still be seen)
- use-first-commit-message-for-new-prs (when otherwise in conflict with use of PR templates)

Comment (tylaing) [reply include/rapidjson/reader.h:156]: Oh well actually, the lines above this are using spaces, whereas you changed line 155 to use a tab and the new line 156 uses a tab.

File: include/rapidjson/reader.h
Comment (thomaslgdskyboxlabs) [inline include/rapidjson/reader.h:1047]: same indent issues

**Review by thomaslgdskyboxlabs — CHANGES_REQUESTED** [review]
good to go after indent fixes

File: include/rapidjson/reader.h
Comment (kawhites) [reply include/rapidjson/reader.h:1047]: This looks normal locally, I'm not sure why it's showing like this on github??

File: include/rapidjson/reader.h
Comment (tylaing) [inline include/rapidjson/reader.h:960]: minor: should use spaces rather than tabs on these 3 lines.

File: include/rapidjson/reader.h
Comment (tylaing) [inline include/rapidjson/reader.h:1046]: minor: should use spaces rather than tabs on this 1 line.

**Review by tylaing — APPROVED** [review]
Approved given there are related automated tests in another PR that show this parses with parity down to the loaded multiline strings matching.

Couple minor whitespace issues that probably should be cleaned up if we are to send this to Tencent, which would be good to do.

**Review by tylaing — APPROVED** [review]

**Review by maxbouchez — COMMENTED** [review]
RapidJson comes with built-in unit and performance tests. We should leverage those test suites to make sure we aren't unintentionally causing issues.
`readertest.cpp` looks like a good place to write a few unit tests to make sure multiline works as intended

**Review by thomaslgdskyboxlabs — APPROVED** [review]

File: include/rapidjson/reader.h
Comment (maxbouchez) [inline include/rapidjson/reader.h:960]: Although I understand this logic might be needed to solve our immediate needs it might make it hard for us to integrate back with Tencent's official repository.
Do we have a work item to revisit this implementation later and build a maintenance plan?

**Review by maxbouchez — APPROVED** [review]
I'm not going to block this PR, but we should make sure to revisit this later

**Pull request merged**
